### PR TITLE
feat(telemetry): add telemetry wrapper for cli

### PR DESCRIPTION
**Pull request by matthieu-crouzet**
## Proposed change
Add telemetry wrapper for cli

## Related issues

- :rocket: Feature #2191

### CODE REVIEW

File: packages/@o3r/workspace/src/cli/set-version.cts [hunk 1]
```diff
@@ -46,20 +46,32 @@ program
...
+      } else {
+        logger.debug(`No change in ${pathWithContent.path}`);
+      }
+    });
+}
+
+void (async () => {
+  try {
+    const { createCliWithMetrics } = await import('@o3r/telemetry');
+    return createCliWithMetrics('@o3r/workspace:set-version', cliFn, { logger, preParsedOptions: options });
```
Comment (matthieu-crouzet) [inline packages/@o3r/workspace/src/cli/set-version.cts:72]: example made once, I will wait for the approval to make the other ones in another PR

File: packages/@o3r/telemetry/package.json [hunk 2]
```diff
@@ -22,11 +22,18 @@
     "@angular-devkit/architect": "~0.1802.0",
     "@angular-devkit/core": "~18.2.0",
     "@angular-devkit/schematics": "~18.2.0",
+    "minimist": "^1.2.6",
```
Comment (fpaul-1A) [inline packages/@o3r/telemetry/package.json:25]: should this be installed with the `ng-add` of `@o3r/telemetry`?

File: packages/@o3r/telemetry/src/cli/index.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
+import { existsSync, readFileSync } from 'node:fs';
+import path from 'node:path';
+import type { Opts as MinimistOptions } from 'minimist';
+import { getEnvironmentInfo } from '../environment';
+import { type CliMetricData, sendData as defaultSendData, type SendDataFn } from '../sender';
+
+/** Simple Logger interface */
+interface Logger {
```
Comment (kpanot) [inline packages/@o3r/telemetry/src/cli/index.ts:8]: Can get the interface from `@o3r/core`

File: packages/@o3r/telemetry/src/cli/index.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  error: (message: string) => void;
+  /** Error message to display */
+  warn: (message: string) => void;
+  /** Information message to display */
+  info: (message: string) => void;
+  /** Debug message message to display */
+  debug: (message: string) => void;
+}
+
+interface CliWrapperOptions {
```
Comment (kpanot) [inline packages/@o3r/telemetry/src/cli/index.ts:19]: Missing comment (as it is part of exported type `CliWrapper`)

File: packages/@o3r/telemetry/package.json (hunk 2 above)
Comment (kpanot) [inline packages/@o3r/telemetry/package.json:25]: Should be flagged as `optional`

File: packages/@o3r/workspace/src/cli/set-version.cts [hunk 5]
```diff
@@ -46,20 +46,32 @@ program
...
+  } catch {
```
Comment (kpanot) [inline packages/@o3r/workspace/src/cli/set-version.cts:73]: It also catch the exception from `createCliWithMetrics` (such as the lack of `minimist`).
I don't know if it is something you want to catch

File: packages/@o3r/telemetry/src/cli/index.ts (hunk 3 above)
Comment (matthieu-crouzet) [reply packages/@o3r/telemetry/src/cli/index.ts:8]: As discussed, we can't as core depends on telemetry

File: packages/@o3r/telemetry/package.json (hunk 2 above)
Comment (matthieu-crouzet) [reply packages/@o3r/telemetry/package.json:21]: as discussed moved as dep

Comment (matthieu-crouzet) [reply packages/@o3r/telemetry/package.json:21]: as discussed moved as dep

File: packages/@o3r/workspace/src/cli/set-version.cts (hunk 5 above)
Comment (matthieu-crouzet) [reply packages/@o3r/workspace/src/cli/set-version.cts:73]: as discussed moved as dep to avoid this kind of issue

Comment (kpanot) [reply packages/@o3r/workspace/src/cli/set-version.cts:73]: Not exactly, `minimist` is now always provided so cannot raise error but an exception can still be raised by other part of the code (an `fs` issue for example).
The point remain valid

**Review by kpanot — DISMISSED** [review]

**Review by fpaul-1A — APPROVED** [review]

**Review by kpanot — APPROVED** [review]

**Pull request merged**